scheduler/__init__.py

```diff
@@ -1,115 +1,283 @@
 import datetime
 import logging
-import json
 import os
-from datetime import datetime,timezone
+from typing import List, Dict
+from datetime import datetime, timezone, timedelta
 import azure.functions as func
 from shared.cosmos_db import was_summarized_today
 from shared.cosmo_data_loader import CosmosDBLoader
-from azure.identity import DefaultAzureCredential
 import requests
 
-process_and_summarize_url = f'{os.environ["WEB_APP_URL"]}/api/SECEdgar/financialdocuments/process-and-summarize'
+# logger setting
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+PROCESS_AND_SUMMARIZE_ENDPOINT = f"{os.environ["WEB_APP_URL"]}/api/SECEdgar/financialdocuments/process-and-summarize"
+EMAIL_ENDPOINT = f'{os.environ["WEB_APP_URL"]}/api/reports/digest'
```

Comment:
@PedroLabrador Using os.environ[...] directly like this will cause a KeyError if WEB_APP_URL is missing.
Use os.getenv() with a default value or error handling.
```python
WEB_APP_URL = os.getenv("WEB_APP_URL", "")
if not WEB_APP_URL:
    raise ValueError("WEB_APP_URL environment variable is not set")

PROCESS_AND_SUMMARIZE_ENDPOINT = f"{WEB_APP_URL}/api/SECEdgar/financialdocuments/process-and-summarize"
EMAIL_ENDPOINT = f"{WEB_APP_URL}/api/reports/digest"
STRIPE_SUBSCRIPTION_ENDPOINT = f"{WEB_APP_URL}/api/subscriptions/<subscription_id>/tiers"

```

Reply:
This won't work raising a ValueError will only re trigger the function, I made a slightly different validation for this

Reply:
Ok I see